test/c/src/cross_serialization_tests.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <cstdbool>
+
+#include "rosidl_runtime_c/message_type_support_struct.h"
+
+#include <rosidl_typesupport_microxrcedds_c/identifier.h>
+#include <rosidl_typesupport_fastrtps_c/identifier.h>
+#include <rosidl_typesupport_introspection_c/identifier.h>
+
+// Workaround for having two coexistent message_type_support_callbacks_t structs
+#define message_type_support_callbacks_t message_type_support_callbacks_fast_t
+#include <rosidl_typesupport_fastrtps_cpp/message_type_support.h>
+#undef message_type_support_callbacks_t
+#define message_type_support_callbacks_t message_type_support_callbacks_xrce_t
+#include <rosidl_typesupport_microxrcedds_c/message_type_support.h>
+#undef message_type_support_callbacks_t
+
+#include "rosidl_typesupport_introspection_c/message_introspection.h"
+#include "rosidl_typesupport_introspection_c/field_types.h"
+
+#include <rosidl_runtime_c/string.h>
+#include <rosidl_runtime_c/string_functions.h>
+
+typedef struct genericSequence {
+  void * data;
+  size_t size;
+  size_t capacity;
+} genericSequence;
+
+template <typename T>
+class CrossSerialization {
+  public:
+
+  #define QUICK_RETURN(x) if(!(x)) return x;
+
+  CrossSerialization(const rosidl_message_type_support_t * type_support_c) :
+    xrce_buffer(nullptr),
+    xrce_size(0),
+    fast_buffer(nullptr),
+    fast_size(0)
+  {
+    xrce_typesupport = get_message_typesupport_handle(type_support_c, rosidl_typesupport_microxrcedds_c__identifier);
+    fast_typesupport = get_message_typesupport_handle(type_support_c, rosidl_typesupport_fastrtps_c__identifier);
+    introspection_typesupport = get_message_typesupport_handle(type_support_c, rosidl_typesupport_introspection_c__identifier);
+
+    xrce_callbacks = static_cast<const message_type_support_callbacks_xrce_t *>(xrce_typesupport->data);
+    fast_callbacks = static_cast<const message_type_support_callbacks_fast_t *>(fast_typesupport->data);
+    introspection = (rosidl_typesupport_introspection_c__MessageMembers*) introspection_typesupport->data;
+  };
+
+  bool check() {
+    bool ret = true;
+    ret &= xrce_typesupport != nullptr;
```

Comment:
Can we use `EXPECT_NE`?

Reply:
No, because we are inside a member method that is nos inheriting from the test class due to the template